src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -104,6 +124,131 @@ public static ImmutableArray<ILocalSymbol> GetDeclaredVariables(this IVariableDe
 
             return arrayBuilder.ToImmutableAndFree();
         }
+
+        /// <summary>
+        /// Get an optional argument name for a named argument to the given <paramref name="dynamicExpression"/> at the given <paramref name="index"/>.
+        /// </summary>
+        /// <param name="dynamicExpression">Dynamic or late bound expression.</param>
+        /// <param name="index">Argument index.</param>
+        public static string GetArgumentName(this IDynamicInvocationExpression dynamicExpression, int index)
```

Comment:
>GetArgumentName [](start = 29, length = 15)

For all new public methods, please add targeted tests to cover scenarios with an invalid input, i.e. when validation fails and we throw.